lib/salus/scanners/osv/go_osv.rb

```diff
@@ -0,0 +1,119 @@
+require 'salus/scanners/osv/base'
+
+module Salus::Scanners::OSV
+  class GoOSV < Base
+    class SemVersion < Gem::Version; end
+    class SemDependency < Gem::Dependency; end
+
+    EMPTY_STRING = "Not Found".freeze
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "LOW".freeze
+
+    def should_run?
+      @repository.go_sum_present?
+    end
+
+    def find_dependencies
+      # Return a map of dependency name and versions used by the project
+      all_dependencies = []
+      chosen_dependencies = {}
+
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      File.foreach(go_sum_path).each("=\n") do |line|
+        line = line.strip
+        next if line.empty?
+
+        go_sum_regex = %r{(?<namespace>(.*)(?!/go\.mod))/(?<name>[^\s]*)
+        (\s)*(?<version>(.*))(\s)*h1:(?<checksum>(.*))}x
+
+        if (matches = line.match(go_sum_regex))
+          all_dependencies.append(
+            {
+              "name": (matches[:namespace] + "/" + matches[:name]).to_s,
+              "version": (matches[:version]).to_s.gsub(%r{/go.mod}, '').strip
+            }
+          )
+        end
+      end
+
+      # Pick specific version of dependencies
+      # If multiple versions of dependencies are found then pick the max version to mimic MVS
+      # https://go.dev/ref/mod#minimal-version-selection
+      all_dependencies.each do |deps|
+        lib = deps[:name]
+        version = deps[:version].to_s.gsub('v', '').gsub('+incompatible', '')
```

Comment:
Could there be an edge case where `deps[:version]` looks like `v1.2.3vxyz`, which would break the line above?

Would it work if you just get rid of the first character if it's `v`?

Reply:
Yeah I think switching to just replacing first char if it's 'v' that might be better.